src/kern/npf_pptp_gre.h

```diff
@@ -0,0 +1,61 @@
+/*-
+ * Copyright (c) 2019 Alex Kiselev alex@therouter.net
+ * All rights reserved.
+ *
+ * This material is based upon work partially supported by The
+ * NetBSD Foundation under a contract with Mindaugas Rasiukevicius.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE NETBSD FOUNDATION, INC. AND CONTRIBUTORS
+ * ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE FOUNDATION OR CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/* 
+ * File:   npf_gre.h
+ * Author: alexk
+ *
+ * Created on June 19, 2019, 12:12 PM
+ */
+
+#ifndef NPF_PPTP_GRE_H
+#define NPF_PPTP_GRE_H
+
+#include "stand/cext.h"
+
+/*
+ * PPTP GRE header
+ */
+struct pptp_gre_hdr {
```

Comment:
I can't remove this header since npf_cache_all() also uses it.
But I left only the pptp_gre_hdr_t in the header and remove reference to it from npf_connkey.c.

npf_cache_all() :
		/* Cache: layer 4 - GRE. */
		npc->npc_l4.hdr = nbuf_advance(nbuf, hlen, sizeof(pptp_gre_hdr_t));

Reply:
Please, see my comment about npf_cache_all() which is still using this header.